src/components/tasks/TaskStatusDropdown.tsx

```diff
@@ -104,26 +116,55 @@ export default function TaskDropDown({
         onChange(payload);
         setMessage('');
     };
+
+    type TaskStatusItem = [string, string];
+
+    interface TaskStatusSelectProps {
+        newStatus: string;
+        handleChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+        taskStatus: TaskStatusItem[];
+        testIdPrefix?: string;
+    }
+
+    const TaskStatusSelect = ({
+        newStatus,
+        handleChange,
+        taskStatus,
+        testIdPrefix = 'task-status',
+    }: TaskStatusSelectProps) => (
+        <select
+            className={styles.taskStatusUpdate}
+            data-testid={testIdPrefix}
+            name="status"
+            onChange={handleChange}
+            value={newStatus}
+        >
+            {taskStatus.map(([name, status]: TaskStatusItem) => (
+                <option
+                    data-testid={`${testIdPrefix}-${name}`}
+                    key={status}
+                    value={status}
+                >
+                    {beautifyStatus(name)}
+                </option>
+            ))}
+        </select>
+    );
```

Comment:
- do not define components inside components, this is an anti-pattern and will cause this component to be re-declared causing the component to be remounted even if something is not changed, either create a new file or declare this component above the current component.